scripts/hotfixes/kdump/hotfix.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+set -euo pipefail
+
+workdir=$(dirname $0)
+
+if [[ $(hostname) == *-pit ]]; then
+    # Exclude ncn-m001 if this is ran from the pit.
+    readarray -t NCNS < <(conman -q | grep -v m001 | sort -u | awk -F - '{print $1"-"$2}')
+else
+    readarray -t NCNS < <(grep -oP 'ncn-\w\d+' /etc/hosts | sort -u)
+fi
+
+for ncn in "${NCNS[@]}"; do
+    printf "Uploading hotfix files to $ncn:/srv/cray/scripts/common/ ... "
+    scp ${workdir}/create-kdump-artifacts.sh ${ncn}:/srv/cray/scripts/common/create-kdump-artifacts.sh >/dev/null
+    scp ${workdir}/dracut-lib.sh ${ncn}:/srv/cray/scripts/common/dracut-lib.sh >/dev/null
+    echo "Done" 
+done
+
+printf "Running updated create-kdump-artifacts.sh script on [${#NCNS[@]}] NCNs ... "
+pdsh -b -w "$(printf '%s,' "${NCNS[@]}")" '/srv/cray/scripts/common/create-kdump-artifacts.sh > /var/log/metal-kdump-hotfix.log 2>/var/log/metal-kdump-hotfix.error.log'
```

Comment:
Same comment here about handling the case where not all NCNs are actually online.  I believe pdsh will exit with non-zero in that case.